distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/JvmOptionsParser.java

```diff
@@ -53,82 +55,88 @@
      * The main entry point. The exit code is 0 if the JVM options were successfully parsed, otherwise the exit code is 1. If an improperly
      * formatted line is discovered, the line is output to standard error.
      *
-     * @param args the args to the program which should consist of a single option, the path to the JVM options
+     * @param args the args to the program which should consist of a single option, the path to ES_PATH_CONF
      */
     public static void main(final String[] args) throws InterruptedException, IOException {
         if (args.length != 1) {
-            throw new IllegalArgumentException("expected one argument specifying path to jvm.options but was " + Arrays.toString(args));
+            throw new IllegalArgumentException("expected one argument specifying path to ES_PATH_CONF but was " + Arrays.toString(args));
         }
+
+        final Stream<Path> jvmOptionsFiles;
```

Comment:
I found this a little hard to grok, so I whipped up a non-stream version, just FYI.

```java
final List<Path> jvmOptionsFiles = new ArrayList<>();
jvmOptionsFiles.add(Paths.get(args[0], "jvm.options"));

final File jvmOptionsDirectory = Paths.get(args[0], "jvm.options.d").toFile();
if (jvmOptionsDirectory.isDirectory()) {
    final File[] files = jvmOptionsDirectory.listFiles();
    Arrays.sort(files, Comparator.comparing(File::getName));

    for (File file : files) {
        jvmOptionsFiles.add(file.toPath());
    }
}
```

Reply:
Okay I pushed something else, what do you think now?